src/js/tech/html5.js

```diff
@@ -282,6 +293,33 @@ class Html5 extends Tech {
     this.textTracks().removeTrack_(e.track);
   }
 
+  removeOldTextTracks() {
+    // This will loop over the texttracks and check if they are still used
+    // If not, they will be removed from the emulated list
+    let removeTracks = [];
+
+    for (let i = 0; i < this.textTracks().length; i++) {
+      let techTrack = this.textTracks()[i];
+
+      let found = false;
+      for (let j = 0; j < this.el().textTracks.length; j++) {
+        if (this.el().textTracks[j] === techTrack) {
```

Comment:
we should also probably cache a reference to the native text track and possibly also do a nullcheck around it as well. Since it is conceivably possible to have an html5 browser that doesn't have text track support.